src/vaults/ProofsFetcher.ts

```diff
@@ -0,0 +1,29 @@
+import { getAllProofs, Proofs } from "../utils/getCurrentOnChainDistribution";
+import { EpochConfig } from "../ages";
+import { getEpochFromId } from "../utils/timestampToEpoch";
+
+export interface ProofsFetcherInterface {
+  fetchProofs: (address: string, epochToId?: string) => Promise<Proofs[]>;
+  getEpochFromId: (epochId: string) => EpochConfig;
+}
+
+export default class ProofsFetcher implements ProofsFetcherInterface {
+  async fetchProofs(address: string, epochToId?: string): Promise<Proofs[]> {
+    const proofs = getAllProofs()
+      .reverse()
+      .filter((proofs) => !!proofs.proofs[address.toLowerCase()]?.amount);
+    if (epochToId) {
+      const epochConfig = getEpochFromId(epochToId);
+      if (!epochConfig) throw Error(`Invalid epoch id ${epochToId}`);
+      const epochIndex = proofs.findIndex((proof) => proof.epoch === epochConfig.id);
+      if (epochIndex === -1) throw Error(`No MORPHO distributed for the vault ${address} in epoch ${epochToId}`);
```

Comment:
This error should also be thrown in the case `epochToId === undefined` and `proofs.length === 0`